src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1550,6 +1554,8 @@ private async ValueTask<(Stream, TransportContext?)> ConnectAsync(HttpRequestMes
 
                 case HttpConnectionKind.Proxy:
                     stream = await ConnectToTcpHostAsync(_proxyUri!.IdnHost, _proxyUri.Port, request, async, cancellationToken).ConfigureAwait(false);
+                    // remoteEndPoint is returned for diagnostic purposes.
+                    remoteEndPoint = GetRemoteEndPoint(stream);
```

Comment:
I'm no longer checking for any telemetry || metrics being enabled here, since `GetRemoteEndpoint` is relatively cheap to run per-connection and likely almost as expensive as the check would be.